exchanges/binance/binance_wrapper.go

```diff
@@ -545,6 +549,26 @@ func (b *Binance) SubmitOrder(s *order.Submit) (order.SubmitResponse, error) {
 	}
 	submitOrderResponse.IsOrderPlaced = true
 
+	// create fee, cost and rate to fill response
+	var fee, cost, priceTemp float64
+	if len(response.Fills) > 0 {
+		for _, tr := range response.Fills {
+			submitOrderResponse.Trades = append(submitOrderResponse.Trades, order.TradeHistory{
+				Price:    tr.Price,
+				Amount:   tr.Qty,
+				Fee:      tr.Commission,
+				FeeAsset: tr.CommissionAsset,
+			})
+			fee += tr.Commission
+			cost += tr.Qty * tr.Price
+			priceTemp += tr.Price
+		}
+
+		submitOrderResponse.Rate = priceTemp / float64(len(response.Fills)) // average rate
```

Comment:
if u buy 4 bitcoins at 10000 and 1 bitcoin at 9900. the average price should be 9980 rather than 9950. i know this is a very exaggerated example (usually the price spreads would be lower) but the effect of that can build up over time or affect strategies and arbiterage which require precision. so instead ud have to do something like: (price*quantity +price*quantity +...+price*quantity)/total quantity <- should give the exact average price

Reply:
You are right, moreover the price can be calculated even easier CumulativeQuoteQty / ExecutedQty
Since we are already returning an slice of Trades, price and fee can be calculated on the client side. This eliminates the need for these fields.